### PR TITLE
Add some UT cases for rflash and rinv 

### PR DESCRIPTION
**Pull request by whowutwut**
These UT test cases check the rflash option parsing code and Resolves **testcase_requested** for #4053 

Here's the run of the UT testcases: 
```
------START::rflash_check::Time:Wed Nov  8 12:50:55 2017------

RUN:rflash mid05tor12cn05 --check [Wed Nov  8 12:50:55 2017]
ElapsedTime:2 sec
RETURN rc = 0
OUTPUT:
mid05tor12cn05: BMC Firmware Product: ibm-v1.99.10-113-g65edf7d-r8-0-g713d86d (Active)*
mid05tor12cn05: HOST Firmware Product: IBM-witherspoon-ibm-OP9_v1.19_1.112 (Active)*
mid05tor12cn05: HOST Firmware Product: -- additional info: buildroot-2017.8-8-g5e23247
mid05tor12cn05: HOST Firmware Product: -- additional info: capp-ucode-p9-dd2-v2
mid05tor12cn05: HOST Firmware Product: -- additional info: hostboot-binaries-edd7653
mid05tor12cn05: HOST Firmware Product: -- additional info: hostboot-e8c85bb
mid05tor12cn05: HOST Firmware Product: -- additional info: linux-4.13.7-openpower1-p1c46946
mid05tor12cn05: HOST Firmware Product: -- additional info: machine-xml-eee619a
mid05tor12cn05: HOST Firmware Product: -- additional info: occ-cff91e5
mid05tor12cn05: HOST Firmware Product: -- additional info: op-build-v1.7-1507-g9a24243
mid05tor12cn05: HOST Firmware Product: -- additional info: petitboot-v1.6.1-p1d8f7a
mid05tor12cn05: HOST Firmware Product: -- additional info: sbe-ec9a99e
mid05tor12cn05: HOST Firmware Product: -- additional info: skiboot-v5.9-rc4
CHECK:rc == 0	[Pass]
CHECK:output =~ mid05tor12cn05: BMC Firmware Product	[Pass]
CHECK:output =~ mid05tor12cn05: HOST Firmware Product	[Pass]
CHECK:rc == 0	[Pass]
 
------END::rflash_check::Passed::Time:Wed Nov  8 12:50:57 2017 ::Duration::2 sec------
------START::rflash_invalid_activate::Time:Wed Nov  8 12:50:57 2017------

RUN:rflash mid05tor12cn05 -a /tmp/abc123.tz [Wed Nov  8 12:50:57 2017]
ElapsedTime:0 sec
RETURN rc = 1
OUTPUT:
mid05tor12cn05: Error: Invalid option specified with -a: /tmp/abc123.tz 
CHECK:rc == 1	[Pass]
CHECK:output =~ Error: Invalid option	[Pass]
 
------END::rflash_invalid_activate::Passed::Time:Wed Nov  8 12:50:57 2017 ::Duration::0 sec------
------START::rflash_invalid_activate_and_delete::Time:Wed Nov  8 12:50:57 2017------

RUN:rflash mid05tor12cn05 -a 123 -d 123 [Wed Nov  8 12:50:57 2017]
ElapsedTime:1 sec
RETURN rc = 1
OUTPUT:
mid05tor12cn05: Error: Multiple options specified is not supported.  Options specified: -a -d
CHECK:rc == 1	[Pass]
CHECK:output =~ Error: Multiple options specified is not supported	[Pass]
 
------END::rflash_invalid_activate_and_delete::Passed::Time:Wed Nov  8 12:50:58 2017 ::Duration::1 sec------
------START::rflash_invalid_id::Time:Wed Nov  8 12:50:58 2017------

RUN:rflash mid05tor12cn05 -a 123 [Wed Nov  8 12:50:58 2017]
ElapsedTime:1 sec
RETURN rc = 1
OUTPUT:
mid05tor12cn05: Error: Invalid ID provided to activate. Use the -l option to view valid firmware IDs.
CHECK:rc == 1	[Pass]
CHECK:output =~ Error: Invalid ID provided to activate	[Pass]
 
------END::rflash_invalid_id::Passed::Time:Wed Nov  8 12:50:59 2017 ::Duration::1 sec------
------START::rflash_invalid_multiple_id_3::Time:Wed Nov  8 12:50:59 2017------

RUN:rflash mid05tor12cn05 -a 123 abc asdbas [Wed Nov  8 12:50:59 2017]
ElapsedTime:1 sec
RETURN rc = 1
OUTPUT:
mid05tor12cn05: Error: More than one firmware specified is not supported.
CHECK:rc == 1	[Pass]
CHECK:output =~ Error: More than one firmware specified is not supported	[Pass]
 
------END::rflash_invalid_multiple_id_3::Passed::Time:Wed Nov  8 12:51:00 2017 ::Duration::1 sec------
------START::rflash_invalid_multiple_id_2::Time:Wed Nov  8 12:51:00 2017------

RUN:rflash mid05tor12cn05 -a asdbas 123 [Wed Nov  8 12:51:00 2017]
ElapsedTime:0 sec
RETURN rc = 1
OUTPUT:
mid05tor12cn05: Error: More than one firmware specified is not supported.
CHECK:rc == 1	[Pass]
CHECK:output =~ Error: More than one firmware specified is not supported	[Pass]
 
------END::rflash_invalid_multiple_id_2::Passed::Time:Wed Nov  8 12:51:00 2017 ::Duration::0 sec------
------START::rflash_invalid_node::Time:Wed Nov  8 12:51:00 2017------

RUN:rflash -a 221d9020 [Wed Nov  8 12:51:00 2017]
ElapsedTime:0 sec
RETURN rc = 1
OUTPUT:
Error: Invalid nodes and/or groups in noderange: 221d9020
CHECK:rc == 1	[Pass]
CHECK:output =~ Error: Invalid nodes and/or groups in noderange: 221d9020	[Pass]
 
------END::rflash_invalid_node::Passed::Time:Wed Nov  8 12:51:00 2017 ::Duration::0 sec------
------Total: 7 , Failed: 0------
```

### CODE REVIEW

**Comment by whowutwut** [comment]
The 2nd commit here creates UT for #4236 and if we run it today, proves that it finds the bugs.  

Output of the 2nd set of testcases 
```
------START::rinv_check_active_fw::Time:Wed Nov  8 13:25:44 2017------

RUN:rinv mid05tor12cn05 firm | grep -i ibm [Wed Nov  8 13:25:44 2017]
ElapsedTime:2 sec
RETURN rc = 0
OUTPUT:
mid05tor12cn05: BMC Firmware Product: ibm-v1.99.10-113-g65edf7d-r8-0-g713d86d (Active)*
mid05tor12cn05: HOST Firmware Product: IBM-witherspoon-ibm-OP9_v1.19_1.112 (Active)*
CHECK:rc == 0	[Pass]
CHECK:output =~ mid05tor12cn05: BMC Firmware Product	[Pass]
CHECK:rc == 0	[Pass]
CHECK:output =~ mid05tor12cn05: HOST Firmware Product	[Pass]
CHECK:rc == 0	[Pass]
 
------END::rinv_check_active_fw::Passed::Time:Wed Nov  8 13:25:46 2017 ::Duration::2 sec------
------START::rinv_check_active_fw_count::Time:Wed Nov  8 13:25:46 2017------

RUN:rinv mid05tor12cn05 firm | grep -i ibm [Wed Nov  8 13:25:46 2017]
ElapsedTime:3 sec
RETURN rc = 0
OUTPUT:
mid05tor12cn05: BMC Firmware Product: ibm-v1.99.10-113-g65edf7d-r8-0-g713d86d (Active)*
mid05tor12cn05: HOST Firmware Product: IBM-witherspoon-ibm-OP9_v1.19_1.112 (Active)*
CHECK:rc == 0	[Pass]
 
RUN:rinv mid05tor12cn05 firm | grep -i ibm | wc -l [Wed Nov  8 13:25:49 2017]
ElapsedTime:2 sec
RETURN rc = 0
OUTPUT:
2
CHECK:output =~ 2	[Pass]
CHECK:rc == 0	[Pass]
 
------END::rinv_check_active_fw_count::Passed::Time:Wed Nov  8 13:25:51 2017 ::Duration::5 sec------
------START::rinv_check_active_fw_verbose_count::Time:Wed Nov  8 13:25:51 2017------

RUN:rinv mid05tor12cn05 firm -V | grep -i Active [Wed Nov  8 13:25:51 2017]
ElapsedTime:4 sec
RETURN rc = 0
OUTPUT:
mid05tor12cn05: [briggs01]: BMC Firmware Product: ibm-v1.99.10-113-g65edf7d-r3-0-g9e4f715 (Active)*
mid05tor12cn05: [briggs01]: BMC Firmware Product: ibm-v1.99.10-113-g65edf7d-r8-0-g713d86d (Active)*
mid05tor12cn05: [briggs01]: HOST Firmware Product: IBM-witherspoon-ibm-OP9_v1.19_1.68 (Active)*
mid05tor12cn05: [briggs01]: HOST Firmware Product: IBM-witherspoon-ibm-OP9_v1.19_1.112 (Active)*
CHECK:rc == 0	[Pass]
 
RUN:rinv mid05tor12cn05 firm -V | grep -i Active | wc -l [Wed Nov  8 13:25:55 2017]
ElapsedTime:3 sec
RETURN rc = 0
OUTPUT:
4
CHECK:rc == 0	[Pass]
CHECK:rc == 0	[Pass]
CHECK:output =~ 2	[Failed]
 
------END::rinv_check_active_fw_verbose_count::Failed::Time:Wed Nov  8 13:25:58 2017 ::Duration::7 sec------
------Total: 3 , Failed: 1------
```

**Comment by whowutwut** [comment]
@hu-weihua can you take a review of this PR and let me know your comments.

File: xCAT-test/autotest/testcase/UT_openbmc/rflash_cases0 [hunk 1]
```diff
@@ -0,0 +1,47 @@
+start:rflash_check
+os:Linux
+mgt:openbmc
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/UT_openbmc/rflash_cases0]: The specific key words ``xcattest`` supported are ``os``, ``arch`` and ``hcp``.  There is not ``mgt``.  I think the key word you wanted to use in line 3 is ``hcp``, right?

File: xCAT-test/autotest/testcase/UT_openbmc/rflash_cases0 [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+cmd:rflash $$CN --check
+check:rc==0
+check:output=~$$CN: BMC Firmware Product
+check:output=~$$CN: HOST Firmware Product
+check:rc==0
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/UT_openbmc/rflash_cases0]: Check ``rc==0`` one time is enough. Line 8 can be deleted.

File: xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0 [hunk 3]
```diff
@@ -0,0 +1,39 @@
+start:rinv_check_active_fw
+description: Ensure that there is only 1 active BMC and 1 Active PNOR
+os:Linux
+mgt:openbmc
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0]: This line should be changed to ``hcp:openbmc``.

File: xCAT-test/autotest/testcase/UT_openbmc/rflash_cases0 [hunk 4]
```diff
@@ -0,0 +1,47 @@
+start:rflash_check
+os:Linux
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/UT_openbmc/rflash_cases0]: It is better to add ``description`` for every test case.

File: xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0 [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+cmd:rinv $$CN firm | grep -i ibm
+check:rc==0
+check:output=~$$CN: BMC Firmware Product
+check:rc==0
+check:output=~$$CN: HOST Firmware Product
+check:rc==0
+end
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0]: I am not sure how does this case to guarantee ``there is only 1 active BMC and 1 Active PNOR``.  If there are 4 firmware active, I thinks ``rc==0`` and 
``check:output=~$$CN: BMC Firmware Product`` will all pass.

File: xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0 [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+start:rinv_check_active_fw_count 
+description: Ensure that there are only 2 active firmware (if previous test is OK)
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0]: Up to now, there is not mechanism in ``xcattest`` or other test framework script to guarantee the dependency between cases.  So it is better to keep the independence of each case.

File: xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0 [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+os:Linux
+mgt:openbmc
+# show the output
+cmd:rinv $$CN firm | grep -i ibm 
+check:rc==0
+# check based on wc 
+cmd:rinv $$CN firm | grep -i ibm | wc -l
+check:output=~2
+check:rc==0
+end
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0]: As I know, the string ``(Active)*`` is written by xcat code. If there is any error in the code, for example, if there are 2 bmc firmware and 2 pnor firmware in the BMC, but xcat code marked the 2 bmc firmware as active accidentally, above check (line 21 and 22) will also pass.  So maybe just check `` there are only 2 active firmware`` is not enough.

File: xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0 [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+# show the output
+cmd:rinv $$CN firm -V | grep -i Active 
+check:rc==0
+# validate test cased on word count
+cmd:rinv $$CN firm -V | grep -i Active | wc -l
+check:rc==0
+check:rc==0
+check:output=~2
+check:rc==0
+end
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0]: I think all above 3 cases are written for issue #4236,  I think all check logic should be include in one case. 

How about code like below:
```
start:rinv_check_active_fw_count
description: Ensure that there is only 2 active firmware, one for bmc and one for pnor
hcp:openbmc
cmd: rinv $$CN firm | tee /tmp/rinv_check_active_fw_count.output
check:rc==0
cmd: grep -i ibm /tmp/rinv_check_active_fw_count.output| grep -i 'HOST Firmware Product' | grep -i 'Active' |wc -l
check:rc==0
check:output=~1
cmd: grep -i ibm /tmp/rinv_check_active_fw_count.output | grep -i 'BMC Firmware Product'|grep -i 'Active' |wc -l
check:rc==0
check:output=~1
cmd : rm -rf /tmp/rinv_check_active_fw_count.output
check:rc==0
end

start:rinv_check_active_fw_verbose_count
description: Ensure that there is only 2 active firmware, one for bmc and one for pnor (in verbose mode) (Issue #4236)
hcp:openbmc
cmd: rinv $$CN firm -V| tee /tmp/rinv_check_active_fw_verbose_count.output
check:rc==0
cmd: grep -i ibm /tmp/rinv_check_active_fw_verbose_count.output| grep -i 'HOST Firmware Product' | grep -i 'Active' |wc -l
check:rc==0
check:output=~1
cmd: grep -i ibm /tmp/rinv_check_active_fw_verbose_count.output | grep -i 'BMC Firmware Product'|grep -i 'Active' |wc -l
check:rc==0
check:output=~1
cmd : rm -rf /tmp/rinv_check_active_fw_verbose_count.output
check:rc==0
end
```
The output looks like below:
```
[root@c910f03c11k08 tmp]# xcattest -f tmp.conf  -t rinv_check_active_fw_count,rinv_check_active_fw_verbose_count
xCAT automated test started at Fri Nov 10 02:27:24 2017
******************************
loading Configure file
******************************
Varible:
    CN = f6u17
    HCP = openbmc
******************************
Initialize xCAT test environment by definition in configure file
******************************
Detecting: OS = Linux
Detecting: ARCH = ppc64le
******************************
loading test cases
******************************
To run:
rinv_check_active_fw_count
rinv_check_active_fw_verbose_count
******************************
Start to run test cases
******************************
------START::rinv_check_active_fw_count::Time:Fri Nov 10 02:27:25 2017------

RUN:rinv f6u17 firm | tee /tmp/rinv_check_active_fw_count.output [Fri Nov 10 02:27:25 2017]
ElapsedTime:4 sec
RETURN rc = 0
OUTPUT:
f6u17: HOST Firmware Product: IBM-witherspoon-ibm-OP9_v1.19_1.112 (Active)*
f6u17: HOST Firmware Product: -- additional info: buildroot-2017.8-8-g5e23247
f6u17: HOST Firmware Product: -- additional info: capp-ucode-p9-dd2-v2
f6u17: HOST Firmware Product: -- additional info: hostboot-binaries-edd7653
f6u17: HOST Firmware Product: -- additional info: hostboot-e8c85bb
f6u17: HOST Firmware Product: -- additional info: linux-4.13.7-openpower1-p1c46946
f6u17: HOST Firmware Product: -- additional info: machine-xml-eee619a
f6u17: HOST Firmware Product: -- additional info: occ-cff91e5
f6u17: HOST Firmware Product: -- additional info: op-build-v1.7-1507-g9a24243
f6u17: HOST Firmware Product: -- additional info: petitboot-v1.6.1-p1d8f7a
f6u17: HOST Firmware Product: -- additional info: sbe-ec9a99e
f6u17: HOST Firmware Product: -- additional info: skiboot-v5.9-rc4
f6u17: BMC Firmware Product: ibm-v1.99.10-113-g65edf7d-r8-0-g713d86d (Active)*
CHECK:rc == 0	[Pass]

RUN:grep -i ibm /tmp/rinv_check_active_fw_count.output| grep -i 'HOST Firmware Product' | grep -i 'Active' |wc -l [Fri Nov 10 02:27:29 2017]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:
1
CHECK:rc == 0	[Pass]
CHECK:output =~ 1	[Pass]

RUN:grep -i ibm /tmp/rinv_check_active_fw_count.output | grep -i 'BMC Firmware Product'|grep -i 'Active' |wc -l [Fri Nov 10 02:27:29 2017]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:
1
CHECK:rc == 0	[Pass]
CHECK:output =~ 1	[Pass]

RUN:rm -rf /tmp/rinv_check_active_fw_count.output [Fri Nov 10 02:27:29 2017]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:
CHECK:rc == 0	[Pass]

------END::rinv_check_active_fw_count::Passed::Time:Fri Nov 10 02:27:29 2017 ::Duration::4 sec------
------START::rinv_check_active_fw_verbose_count::Time:Fri Nov 10 02:27:29 2017------

RUN:rinv f6u17 firm -V| tee /tmp/rinv_check_active_fw_verbose_count.output [Fri Nov 10 02:27:29 2017]
ElapsedTime:3 sec
RETURN rc = 0
OUTPUT:
f6u17: [c910f03c11k08]: BMC Firmware Product: ibm-v1.99.10-0-r7-0-gcce0f1b (Active)*
f6u17: [c910f03c11k08]: HOST Firmware Product: IBM-witherspoon-ibm-OP9_v1.19_1.112 (Active)*
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: buildroot-2017.8-8-g5e23247
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: capp-ucode-p9-dd2-v2
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: hostboot-binaries-edd7653
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: hostboot-e8c85bb
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: linux-4.13.7-openpower1-p1c46946
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: machine-xml-eee619a
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: occ-cff91e5
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: op-build-v1.7-1507-g9a24243
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: petitboot-v1.6.1-p1d8f7a
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: sbe-ec9a99e
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: skiboot-v5.9-rc4
f6u17: [c910f03c11k08]: HOST Firmware Product: IBM-witherspoon-redbud-ibm-OP9_v1.19_1.33 (Active)*
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: buildroot-2017.8-6-g319c6e1
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: capp-ucode-9c73e9f
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: hostboot-3d6c541
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: hostboot-binaries-836385d
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: linux-4.13-openpower1-pe8fb223
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: machine-xml-f8aaa73-p5482dba
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: occ-a43395b
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: op-build-450d9d7
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: petitboot-v1.5.1-p36336db
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: sbe-8d90ab2
f6u17: [c910f03c11k08]: HOST Firmware Product: -- additional info: skiboot-v5.8-73-g4a2b8317fd3f-p6faaa2d
f6u17: [c910f03c11k08]: BMC Firmware Product: ibm-v1.99.10-113-g65edf7d-r8-0-g713d86d (Active)*
CHECK:rc == 0	[Pass]

RUN:grep -i ibm /tmp/rinv_check_active_fw_verbose_count.output| grep -i 'HOST Firmware Product' | grep -i 'Active' |wc -l [Fri Nov 10 02:27:32 2017]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:
2
CHECK:rc == 0	[Pass]
CHECK:output =~ 1	[Failed]

RUN:grep -i ibm /tmp/rinv_check_active_fw_verbose_count.output | grep -i 'BMC Firmware Product'|grep -i 'Active' |wc -l [Fri Nov 10 02:27:32 2017]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:
2

RUN:rm -rf /tmp/rinv_check_active_fw_verbose_count.output [Fri Nov 10 02:27:32 2017]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:

------END::rinv_check_active_fw_verbose_count::Failed::Time:Fri Nov 10 02:27:32 2017 ::Duration::3 sec------
------Total: 2 , Failed: 1------

xCAT automated test finished atFri Nov 10 02:27:32 2017
Please check results in the /opt/xcat/bin/../share/xcat/tools/autotest/result/,
and see /opt/xcat/bin/../share/xcat/tools/autotest/result//failedcases.20171110022724 file for failed cases.
see /opt/xcat/bin/../share/xcat/tools/autotest/result//performance.report.20171110022724 file for time consumption
```

Comment (hu-weihua) [reply xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0]: One more thing, if pull request #4279 was merged, the key word ``hcp`` will work. Above output come from new ``xcattest``.

File: xCAT-test/autotest/testcase/UT_openbmc/rflash_cases0 [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+check:output=~Error: More than one firmware specified is not supported
+end 
+
+start:rflash_invalid_node
+cmd: rflash -a 221d9020
+check:rc==1 
+check:output=~Error: Invalid nodes and/or groups in noderange: 221d9020
+end
+
+
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/UT_openbmc/rflash_cases0]: Should we have cases to cover delete more than one ID at same time?

**Comment by hu-weihua** [comment]
Another problem, the sub directory under ``xCAT-test/autotest/testcase/`` are categorized by command. I am not sure if it is a good option to save UT test here.

File: xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0 (hunk 8 above)
Comment (whowutwut) [reply xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0]: Awesome comment! Thank you!

File: xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0 (hunk 7 above)
Comment (whowutwut) [reply xCAT-test/autotest/testcase/UT_openbmc/rinv_cases0]: Yes agree... here is where we need to check against the API data and make sure xCAT is returning the correct thing.  But if we see more than 1, we should flag it as error regardless and maybe Dev can debug further.

**Comment by whowutwut** [comment]
@hu-weihua Regarding the comment above about the directories divided by commands... 

So I wanted to start somewhere instead of changing the format of all testcases, I just created something that I thought worked for my use case.  I am OK to change this as I start to run it or if we get some feedback.  But I want to be able to:
1. Quickly find UT testcases that DO NOT run provisioning 
2. Filter on openbmc/ipmi or different mgmt methods 

If we have a better way or if we start to use it and find it's not right, I will be more then willing to change.

**Comment by whowutwut** [comment]
@hu-weihua Thank you so much for your comments, I have added a few more commits to this PR.

**Comment by hu-weihua** [comment]
@whowutwut, The code of these UT cases are fine for me.  
About where and how to save these UT cases, I do not have better idea now. 
So let merge it first, if we find a better way in future, we can move them then.

**Pull request merged**